internal/host/plugin/repository_host_catalog.go

```diff
@@ -164,3 +181,91 @@ func (r *Repository) ListCatalogs(ctx context.Context, scopeIds []string, opt ..
 	}
 	return hostCatalogs, nil
 }
+
+// getCatalog retrieves the *HostCatalog with the provided id.  If it is not found or there
+// is an problem getting it from the database an error is returned instead.
+func (r *Repository) getCatalog(ctx context.Context, id string) (*HostCatalog, error) {
```

Comment:
Maybe a TODO, but it feels like rather than these two functions it should be a view.

Reply:
I've added a TODO.